inspirehep/modules/recordvalidators/validators/rules.py

```diff
@@ -0,0 +1,402 @@
+    except ValidationError:
+        raise
+    return {}
+
+
+def check_date_present_in_record(record):
+    for field, properties in six.iteritems(SCHEMA_FIELDS_THAT_CAN_CONTAIN_DATE):
+        if field in record:
+            if properties:
+                for prop in properties:
+                    if check_if_field_exists_in_dict_list_x_times(prop, record[field], 1):
+                        # return after 1st occurrence
+                        return {}
+            else:
+                return {}
+    raise ValidationError(key='globalErrors',
+                          message='No date present.')
+
+
+def check_if_journal_title_is_canonical(record):
+    def _check_for_journal_title(field, prop, value, index, errortype):
+        querystring = GET_RESULTS_FOR_FIELD_PROPERTY_QUERYSTRING_TEMPLATE.format(field=field,
+                                                                                 prop=prop,
+                                                                                 value=value)
+        results = get_query_results_if_exceed_limit(querystring, 0)
+        if not results:
+            raise ValidationError(key='/{field}/{index}/{prop}'.format(field=field, index=index, prop=prop),
+                                  message="Journal title '{title}' doesn't exist.'".format(title=value),
+                                  type=errortype)
+    try:
+        check_if_listfield_property_values_are_valid(record=record,
+                                                     field='publication_info',
+                                                     prop='journal_title',
+                                                     checker=_check_for_journal_title)
+    except ValidationError:
+        raise
+    return {}
+
+
+def check_languages_if_valid_iso(record):
+    invalid_languages = {}
+
+    def _check_language(language, index):
+        try:
+            # TODO: does case insensitive lookup. is it ok?
+            pycountry.languages.lookup(language)
+        except LookupError:
+            pass
+            raise ValidationError(key='/languages/{index}'.format(index=index),
+                                  message=FIELD_VALIDATION_TEMPLATE.format(
+                                      field='languages',
+                                      value=language
+                                  ))
+    for i, item in enumerate(record.get('languages', [])):
+        try:
+            _check_language(item, i)
+        except ValidationError as e:
+            invalid_languages.update(e.error)
+
+    if invalid_languages:
+        raise ValidationError(error=invalid_languages)
+    return {}
+
+
+'''
+Warnings Validator Functions
+'''
+
+
+def check_affiliations_if_authors_exist(record):
+    def _check_affiliations_for_author(author, index):
+        if 'affiliations' not in author or len(author['affiliations']) == 0:
```

Comment:
Yes, for the moment leave it without and we see if it blows up. Then we can fix the piece of code that is adding empty lists.